druid/src/widget/view_switcher.rs

```diff
@@ -79,6 +79,9 @@ impl<T: Data, U: PartialEq> Widget<T> for ViewSwitcher<T, U> {
         }
 
         if !old_data.same(data) {
+            if let Some(child) = self.active_child.as_mut() {
```

Comment:
cool, yea, this was definitely a bug.

I don't actually think we should be doing the old_data.same(data) check; we should just entirely delegate this to the child, which can do what it needs. We also shouldn't call `request_paint`; that's also up to the child.

Reply:
@cmyr Thanks for the feedback.
I addressed your comment, and checked that it still works with my code.